### PR TITLE
Adding image tag field

### LINKED ISSUE #5: Image Tags
#### As a user I want to add tags to images.

__Story__:

Our users are loving the ability to add images to our service. However, now
that there are more than a handful of images, it is apparent that we need some
way to filter images by certain categories. In order to do that, we first need
to label images in some way. It is time to add the ability for our users to tag
images when they add them.

__Acceptance criteria__:
- [x] When I add a new image, there is a form field to add tags.
- [x] After I've saved an image with tags, I see those tags displayed with the
  associated image.
- [x] My application uses the acts-as-taggable-on gem to provide the tagging
  functionality.

__Reference material__:
- https://github.com/mbleigh/acts-as-taggable-on

__Note__: You should allow images to have no tags; this sets the stage for a later issue.

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by dhanson11**
Closes #5

### CODE REVIEW

**Comment by dhanson11** [comment]
oops had to fix commit history

File: db/migrate/20190801222403_acts_as_taggable_on_migration.acts_as_taggable_on_engine.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# This migration comes from acts_as_taggable_on_engine (originally 1)
+if ActiveRecord.gem_version >= Gem::Version.new('5.0')
+  class ActsAsTaggableOnMigration < ActiveRecord::Migration[4.2]; end
+else
+  class ActsAsTaggableOnMigration < ActiveRecord::Migration; end
+end
+ActsAsTaggableOnMigration.class_eval do
+  def self.up
+    create_table :tags do |t|
+      t.string :name
```
[the remark this answers is not in the document]
Comment (dhanson11) [reply db/migrate/20190801222403_acts_as_taggable_on_migration.acts_as_taggable_on_engine.rb]: should this be done in the migration or in the model?

[the remark this answers is not in the document]
Comment (JingyuZ) [reply db/migrate/20190801222403_acts_as_taggable_on_migration.acts_as_taggable_on_engine.rb]: We want to put it in migration first, so MySQL won't allow `null` value in db. Validation in model will be helpful too, but this is a different level of validation from db.

**Comment by dhanson11** [comment]
I think its ready to go! ✅

**Pull request merged**